File: sdk/ts/e2e/runner.ts
import { slogx } from '../slogx';

type Mode = 'ws' | 'ci';

const args = process.argv.slice(2);
const getArg = (name: string, fallback?: string) => {
  const idx = args.indexOf(name);
  if (idx === -1) return fallback;
  return args[idx + 1] ?? fallback;
};

const mode = (getArg('--mode', 'ws') as Mode);
const port = parseInt(getArg('--port', '8091') || '8091', 10);
const filePath = getArg('--file', '');
const serviceName = getArg('--service', 'ts-e2e') || 'ts-e2e';
const intervalMs = parseInt(getArg('--interval', '200') || '200', 10);

const prefix = `SDK TS ${mode.toUpperCase()}`;

const sendLogs = () => {
  slogx.info(`${prefix} alpha`, { fixture: 1 });
  slogx.warn(`${prefix} beta`, { fixture: true });
  try {
    throw new Error(`${prefix} error`);
  } catch (err) {
    slogx.error(`${prefix} gamma`, err);
  }
  slogx.debug(`${prefix} delta`, { nested: { ok: true } });
};

const run = async () => {
  await slogx.init({
    isDev: true,
    port,
    serviceName,
    ciMode: mode === 'ci',
    logFilePath: filePath || undefined
  });

  if (mode === 'ws') {
    console.log(`[slogx-e2e] READY ws://localhost:${port}`);
    setInterval(sendLogs, intervalMs);
    return;
  }

  sendLogs();
  setTimeout(() => {
    sendLogs();
  }, 650);

  setTimeout(() => {
    slogx.close();
    process.exit(0);
  }, 900);
};

process.on('SIGTERM', () => {
  slogx.close();
  process.exit(0);
});

process.on('SIGINT', () => {
  slogx.close();
  process.exit(0);
});

run().catch((err) => {
  console.error('[slogx-e2e] Failed to start', err);
  process.exit(1);
});

File: sdk/go/e2e/runner.go
package main

import (
	"flag"
	"fmt"
	"os"
	"os/signal"
	"strings"
	"syscall"
	"time"

	"github.com/binhonglee/slogx/sdk/go/slogx"
)

func emit(prefix string) {
	slogx.Info(fmt.Sprintf("%s alpha", prefix), map[string]any{"fixture": 1})
	slogx.Warn(fmt.Sprintf("%s beta", prefix), map[string]any{"fixture": true})
	slogx.Error(fmt.Sprintf("%s gamma", prefix), fmt.Errorf("%s error", prefix))
	slogx.Debug(fmt.Sprintf("%s delta", prefix), map[string]any{"nested": map[string]any{"ok": true}})
}

func main() {
	mode := flag.String("mode", "ws", "ws|ci")
	port := flag.Int("port", 8093, "port")
	filePath := flag.String("file", "", "ndjson output path")
	service := flag.String("service", "go-e2e", "service name")
	interval := flag.Duration("interval", 200*time.Millisecond, "ws emit interval")
	flag.Parse()

	ciMode := *mode == "ci"

	slogx.Init(slogx.Config{
		IsDev:       true,
		Port:        *port,
		ServiceName: *service,
		CIMode:      &ciMode,
		LogFilePath: *filePath,
		MaxEntries:  1000,
	})

	prefix := fmt.Sprintf("SDK GO %s", strings.ToUpper(*mode))

	stop := make(chan os.Signal, 1)
	signal.Notify(stop, os.Interrupt, syscall.SIGTERM)

	if *mode == "ws" {
		fmt.Printf("[slogx-e2e] READY ws://localhost:%d\n", *port)
		ticker := time.NewTicker(*interval)
		defer ticker.Stop()
		for {
			select {
			case <-ticker.C:
				emit(prefix)
			case <-stop:
				return
			}
		}
	}

	emit(prefix)
	time.Sleep(650 * time.Millisecond)
	emit(prefix)
	time.Sleep(700 * time.Millisecond)
}
